### PR TITLE
Add GitHub Actions workflow for running targets pipeline

### PR DESCRIPTION
**Pull request by program--**
This PR adds a `.github/workflows/generate-catalog.yaml` workflow file that runs the `targets` pipeline through GitHub Actions. This PR also adds a `DESCRIPTION` file to manage project dependencies via `pak` with its lockfile mechanism.

Specifically, this workflow only runs on changes to:
- `_targets.R`
- `DESCRIPTION`
- any file in the `R/` directory
- any file in the `data-raw/` directory

### CODE REVIEW

**Comment by mikejohnson51** [comment]
Let's please update to climateR 0.3.1. to remove opendap.catalog. Thanks!

File: .github/workflows/generate-catalog.yaml [hunk 1]
```diff
@@ -34,6 +34,9 @@ on:
       - 'DESCRIPTION'
       - 'R/**.R'
       - 'data-raw/*'
+  schedule:
+    # Run this workflow at 00:00 UTC every Wednesday
+    - cron: '0 0 * * 3'
```
Comment (mikejohnson51) [inline .github/workflows/generate-catalog.yaml]: We can eventually get away with once a month I think

File: .github/workflows/generate-catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# MIT License
+# Copyright (c) 2021 Eli Lilly and Company
+# Author: William Michael Landau (will.landau at gmail)
```
Comment (mikejohnson51) [inline .github/workflows/generate-catalog.yaml:3]: I would like to avoid dependent licenses if possible?

File: DESCRIPTION [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    sf,
+    rvest,
+    jsonlite,
+    logger,
+    pacman,
+    RNetCDF,
+    data.table,
+    RCurl,
+    AOI,
+    opendap.catalog,
```
Comment (mikejohnson51) [inline DESCRIPTION]: eventual remove

File: .github/workflows/generate-catalog.yaml (hunk 2 above)
Comment (program--) [reply .github/workflows/generate-catalog.yaml:3]: I'm not sure what you mean by dependent licenses? That would suggest R packages dependencies would also need to be avoided 😬

**Pull request merged**